server/src/engine/socket.ts

```diff
@@ -19,13 +19,12 @@ import * as queue from './queue'
 import { clearDiagnostics, sendNotification } from '../server'
 import { EventEmitter } from 'events'
 import { handleCrash, lspCheckResponseHandler } from '../handlers'
-import { getPort } from 'portfinder'
```

Comment:
We used portfinder to ensure that you can start multiple independent VSCode instances. Does that still work, and if so, how?

Reply:
(Each instance would use a fresh port)

Reply:
Last time I looked at this, I thought it was a bug, since portfinder has already been used to find the port passed to the socket. What I can see now is that the socket uses it when it loses its connection to the compiler, to check whether the compiler's process is still running by looking at whether or not the port has become free.

This means that the old code restarted the compiler-process when it died, but the new code does not, which is why the `internalRestart` became unused.

Reply:
It's fixed now

Reply:
Now the new code also restarts it

Reply:
We still use portfinder right, but only in flix.ts right?

Reply:
Yes